lib/travis/build/script/crystal.rb

```diff
@@ -0,0 +1,48 @@
+module Travis
+  module Build
+    class Script
+      class Crystal < Script
+
+        def configure
+          super
+          sh.cmd %q(sudo sh -c 'apt-key adv --keyserver keys.gnupg.net --recv-keys 09617FD37CC06B54')
+          sh.cmd %q(sudo sh -c 'echo "deb http://dist.crystal-lang.org/apt crystal main" > /etc/apt/sources.list.d/crystal.list')
+          sh.cmd %q(sudo sh -c 'apt-get update')
+        end
+
+        def setup
+          super
+
+          [ 'Crystal for Travis-CI is not officially supported, but is community maintained.',
+            'Please file any issues using the following link',
+            '  https://github.com/travis-ci/travis-ci/issues/new?labels=community:crystal',
+            'and mention \`@asterite\`, \`@jhass\`, \`@waj\` and \`@will\` in the issue'
+          ].each { |line| sh.echo(line, ansi: :green) }
+
+          sh.fold 'crystal_install' do
+            sh.echo 'Installing Crystal', ansi: :yellow
+            sh.cmd %q(sudo apt-get install crystal')
```

Comment:
`setup` is invoked after `sudo` is disabled, so this would not work. This needs to be moved to `configure`.